[change:radius] Replace third-party JSONField with Django built-in JSONField

Eeshu-Yadav opened the pull request:
## Checklist

- [x] I have read the [OpenWISP Contributing Guidelines](http://openwisp.io/docs/developer/contributing.html).
- [x] I have manually tested the changes proposed in this pull request.
- [x] I have written new test cases for new code and/or updated existing tests for changes to existing code.
- [ ] I have updated the documentation.

## Reference to Existing Issue

Closes #600.

## Description of Changes

This PR replaces the third-party `jsonfield` library with Django's built-in JSONField. The third-party JSONField library hasn't received updates in 5 years, while Django's built-in JSONField is now mature and actively maintained.

### Changes Made:
- Replaced `from jsonfield import JSONField` with `from django.db.models import JSONField` in models
- Updated `RadiusBatch.user_credentials` and `OrganizationRadiusSettings.sms_meta_data` fields
- Removed `"jsonfield~=3.1.0"` dependency from setup.py
- Added migration `0041_replace_jsonfield_with_django_builtin.py` to handle field type transition
- Updated test migrations to use Django's JSONField

### Testing:
- All existing tests pass with the new implementation
- Verified RadiusBatch and SMS metadata functionality
- Migration tested without data loss

nemesifier commented on openwisp_radius/migrations/0043_alter_organizationradiussettings_sms_meta_data_and_more.py:
```diff
@@ -0,0 +1,52 @@
+                batch.user_credentials = json.loads(batch.user_credentials)
+                batch.save(update_fields=["user_credentials"])
+            except (json.JSONDecodeError, TypeError):
+                # If it's already decoded or invalid JSON, skip it
```
It shouldn't fail. If it fails we must let it fail loudly, unless there can be failure which would be common and in that case we should catch the exception and resolve this automatically.

Why do you think it would fail here?

nemesifier reviewed (changes requested):
Thanks @Eeshu-Yadav, see my comments below.

Eeshu-Yadav commented:
@nemesifier kindly review

Eeshu-Yadav committed:
[change:radius] Replace third-party JSONField with Django built-in JSONField #600

Replaced jsonfield package's JSONField with Django's built-in JSONField
in RadiusBatch.user_credentials and OrganizationRadiusSettings.sms_meta_data.

Rewrote the two historical migrations that referenced jsonfield.fields
(0007_sms_verification, 0009_radbatch_user_credentials_field) to use
django.db.models.JSONField directly, allowing the jsonfield package to
be removed from setup.py.

RadiusBatch.prefix_add now assigns user_credentials as a Python list
directly instead of json.dumps()-ing it first. Migration 0044 converts
existing double-encoded JSON strings in user_credentials to proper JSON
values, iterating with .iterator() for memory efficiency.

Closes #600

nemesifier reviewed (changes requested):
@Eeshu-Yadav is there a reason for which the jsonfield python dependency is not being removed?

Eeshu-Yadav commented:
> @Eeshu-Yadav is there a reason for which the jsonfield python dependency is not being removed?

updated , kindly review
